### ISSUE TITLE
Garbled screen after start (fonts then graphics) in Dracula (unplayable)

### ISSUE THREAD

**Issue by mikerro** [laoo/Felix#54]

**Comment by 42Bastian** [comment]
I checked, it is not a sprite-engine issue. They build the text into a buffer and then draw this sprite.

**Comment by 42Bastian** [comment]
Some info for searching:
Sprite rendering is at $1615
SCB for text sprite is at $497e, SCB for CLS is $168c, SCB for Bram Stoker is $4153
Data for text sprite is at $4999
This is the sprite data to be found at $4999
[D2.zip](https://github.com/laoo/Felix/files/7569447/D2.zip)

**Comment by laoo** [comment]
Thanks, I'll look into it.

**Comment by 42Bastian** [comment]
How can I dump in VS the Lynx memory?

**Comment by laoo** [comment]
@42Bastian I'll write a command for it eventually (hopefully I'll soon be able to take on R/W/X escape functions in lua) but now I'm using an extension called [VSDebugPro](https://marketplace.visualstudio.com/items?itemName=OvidiuIonescu.VSDebugTool). Then I breaking somewhere where mRAM is visible and issue in `VSD Console`:

`dumpmem b:/ram.dat &mRAM[0] 65536`

**Comment by 42Bastian** [comment]
Cool, they have a fake screen and use SUZY do render the text into this fake screen. Later they use this as one big sprite-data block.

**Comment by 42Bastian** [comment]
Ok, found a way to view mRAM, the rendering fails. At $4999 is already garbage. Not sure if wrong data is fetched or using an odd address for the render buffer is the problem.

**Comment by laoo** [comment]
It's hard to debug, but I managed to discover that they are blitting the text using one bit per pixel. Maybe it's buggy. I don't remember if I tested it.

**Comment by 42Bastian** [comment]
I copied the sprite data from Handy and run it in my own program, but no mixed up painting.
Changed VIDBAS to be odd: No problem.
Next is to check if the sprite data in Felix is the same as in Handy.

**Comment by 42Bastian** [comment]
Sprite drawing process does not detect the end of the sprite data and changes to next quadrant and  continues drawing.
At address $2842 is the call to draw the characters. Each is drawn to $4999. Everything works correct until the last line is drawn.
I will try to reproduce this in a simple example.

**Comment by 42Bastian** [comment]
Oh, no. Not Suzy drawing is wrong. In the sprite data appears a "$01" (change quadrant) but it should be a "$02".

**Comment by 42Bastian** [comment]
Hey, after a in-game restart (opt1+pause), the text is correct?!

**Comment by 42Bastian** [comment]
Nope, it is the sound. Steps to reproduce:
- load 
- start game (opt 1)
- disable music (opt 2)
- restart game (opt 1+pause) (music is off!)
- text correct, start game (opt 1)
- enable music (opt 2) => text garbled.

One can see flickering pixels in the game screen while music is active.

Maybe something related to interrupts?

**Comment by laoo** [comment]
Wow... What a plot twist :)
Maybe it would make sense to focus on some debugging facilities...

**Comment by 42Bastian** [comment]
> Wow... What a plot twist :) Maybe it would make sense to focus on some debugging facilities...

Right, currently I modify cpu.c to check the current pc => program breakpoint,
or core.c => data watch point.
I tried VS conditional breakpoints (like state.pc == 0x1634), but this slows execution way to much.
I would be helpful if the Lynx memory would be globally visible.
BTW: If you plan to add breakpoints et al: Bind them to the instance. Handy looses them on reloading.
